libnavigation-util/src/main/java/com/mapbox/navigation/utils/internal/ContextEx.kt

```diff
@@ -0,0 +1,28 @@
+package com.mapbox.navigation.utils.internal
+
+import android.content.Context
+import android.location.LocationManager
+import android.os.Build
+import android.provider.Settings
+
+/**
+ * Check if enabled one of location providers: GPS_PROVIDER or NETWORK_PROVIDER.
+ *
+ * @return true if enabled at least one of location providers
+ */
+fun Context.isLocationProviderEnabled(): Boolean =
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+        val lm = getSystemService(Context.LOCATION_SERVICE) as LocationManager?
+        lm?.isLocationEnabled ?: false
+    } else {
+        try {
+            val mode = Settings.Secure.getInt(
+                contentResolver,
+                Settings.Secure.LOCATION_MODE,
```

Comment:
From docs

```
* @deprecated To check location status, use {@link LocationManager#isLocationEnabled()}. To
*             get the status of a location provider, use
*             {@link LocationManager#isProviderEnabled(String)}.
```

Same for `LOCATION_MODE_OFF`.

I believe we should fallback to [`LocationManager#isProviderEnabled`](https://developer.android.com/reference/android/location/LocationManager#isProviderEnabled(java.lang.String)) as we did in https://github.com/mapbox/mapbox-navigation-android/pull/2902#discussion_r425422250 and do something with the potential `SecurityException` (as implemented here `try {} catch() {}`). What do you think?